Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1923,3 +1932,51 @@ private extension Purchases {
     }
 
 }
+
+// MARK: - Win-Back Offers
+#if !ENABLE_CUSTOM_ENTITLEMENT_COMPUTATION
+@available(iOS 18.0, macOS 15.0, tvOS 18.0, watchOS 11.0, visionOS 2.0, *)
+extension Purchases {
+
+    /**
+     * Returns the win-back offers that the subscriber is eliglble for on the provided product.
+     *
+     * - Parameter product: The product to check for eliglble win-back offers.
+     * - Returns: The win-back offers on the given product that a subscriber is eliglble for.
+     * - Important: Win-back offers are only supported when the SDK is running with StoreKit 2 enabled.
+     */
+    internal func eligibleWinBackOffers(
+        forProduct product: StoreProduct
+    ) async throws -> [WinBackOffer] {
+        return try await self.purchasesOrchestrator.eligibleWinBackOffers(forProduct: product)
+    }
+
+    /**
+     * Returns the win-back offers that the subscriber is eliglble for on the provided product.
+     *
+     * - Parameter product: The product to check for eliglble win-back offers.
+     * - Parameter completion: A completion block that is called with the eligible win-back
+     * offers for the provided product.
+     * - Important: Win-back offers are only supported when the SDK is running with StoreKit 2 enabled.
+     */
+    internal func eligibleWinBackOffers(
+        forProduct product: StoreProduct,
+        completion: @escaping @Sendable (Result<[WinBackOffer], PublicError>) -> Void
```

Comment:
completion is marked as `@Sendable` to avoid warnings below when we dispatch the completion handler on the main actor